src/android/CameraPreviewFragment.java

```diff
@@ -258,6 +278,67 @@ public void hasFlash(HasFlashCallback hasFlashCallback) {
         hasFlashCallback.onResult(camera.getCameraInfo().hasFlashUnit());
     }
 
+    public void startVideoCapture(VideoCallback videoCallback) {
+        if (ActivityCompat.checkSelfPermission(this.getContext(), Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this.getActivity(), new String[]{Manifest.permission.RECORD_AUDIO}, 200);
+        }
+
+        if (recording != null) {
+            recording.stop();
+            recording = null;
+            return;
+        }
+
+        UUID uuid = UUID.randomUUID();
+        String filename = uuid.toString() + ".mp4";
+
+        File videoFile = new File(
+                getContext().getFilesDir(),
+                filename
+        );
+
+        FileOutputOptions outputOptions = new FileOutputOptions.Builder(videoFile).build();
+
+        final Handler handler = new Handler(Looper.getMainLooper());
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                stopVideoCapture();
+            }
+        }, 30000);
+
+        recording = videoCapture.getOutput()
+                .prepareRecording(this.getContext().getApplicationContext(), outputOptions)
+                .withAudioEnabled()
+                .start(ContextCompat.getMainExecutor(this.getContext()), videoRecordEvent -> {
+                    if (videoRecordEvent instanceof VideoRecordEvent.Start) {
+                        videoCallback.onStart(true, null);
+                    } else if (videoRecordEvent instanceof VideoRecordEvent.Finalize) {
+                        VideoRecordEvent.Finalize finalizeEvent = (VideoRecordEvent.Finalize) videoRecordEvent;
+                        handler.removeCallbacksAndMessages(null);
+                        if (finalizeEvent.hasError()) {
+                            int errorCode = finalizeEvent.getError();
+                            Throwable errorCause = finalizeEvent.getCause();
+                            videoCallback.onError(errorCode + " " + errorCause);
+                        } else {
+                            videoCallback.onStop(false, Uri.fromFile(videoFile).toString());
+                            Uri savedUri = finalizeEvent.getOutputResults().getOutputUri();
+                        }
+                        recording = null;
+                    }
+                });
+
+    }
+
+    public void stopVideoCapture() {
+        if (recording != null) {
```

Comment:
I think we have to put this,
```suggestion
        if (recording != null && recording == true) {
```